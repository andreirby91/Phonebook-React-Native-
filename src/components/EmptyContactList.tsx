import React from 'react';
import {StyleSheet, View, Text, TouchableOpacity} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import {RootStackParamList} from '../../App';

type Props = {
  onPress: () => void;
};

const styles = StyleSheet.create({});

const EmptyContactList = ({onPress}: Props) => {
  return (
    <View>
      <Text>No contacts available!</Text>
      <TouchableOpacity onPress={onPress}>
        <Text>Add new contact</Text>
      </TouchableOpacity>
    </View>
  );
};

export default EmptyContactList;
