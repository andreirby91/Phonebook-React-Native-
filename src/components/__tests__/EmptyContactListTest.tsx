import React from 'react';
import renderer from 'react-test-renderer';

import EmptyContactList from '../EmptyContactList';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<EmptyContactList onPress={() => {}} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
