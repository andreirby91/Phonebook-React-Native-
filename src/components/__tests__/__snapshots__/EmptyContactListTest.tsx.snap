// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    No contacts available!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0984e3",
        "borderRadius": 30,
        "height": 50,
        "justifyContent": "center",
        "minWidth": 200,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
        }
      }
    >
      Add new contact
    </Text>
  </View>
</View>
`;
